### ISSUE TITLE
Variable value corruption

### ISSUE THREAD

**Issue by pjtaylor** [SQLPlayer/azure.datafactory.tools#80]
**Describe the bug**
  "value": "@variables('loaddate')" is being corrupted when put through the config change engine

**To Reproduce**
Run the attached pipeline through the config change engine

[ValueSetIssue.zip](https://github.com/SQLPlayer/azure.datafactory.tools/files/5844443/ValueSetIssue.zip)
```
"value": {
    "value": "@variables('loaddate')",
    "type": "Expression"
}
```
is changed to 
```
"value": "@{value=@variables(\u0027loaddate\u0027); type=Expression}"
```
Pipeline is deployed but shows an error in the GUI

**Expected behaviour**
```
"value": {
    "value": "@variables('loaddate')",
    "type": "Expression"
}
```
is changed to 
```
"value": {
    "value": "@variables(\u0027loaddate\u0027)",
    "type": "Expression"
}
```
This is what is currently happening in a similar pipeline which is being deployed successfully.

**Screenshots**
N/A

**Version of PowerShell module of:**

ModuleType Version    Name                                ExportedCommands                                                                                                                    
---------- -------    ----                                ----------------                                                                                                                    
Script     0.30.0     azure.datafactory.tools
Script     1.11.3     Az.DataFactory                 

Major         : 5
Minor         : 1
Build         : 18362
Revision      : 1171
MajorRevision : 0
MinorRevision : 1171

**Comment by NowinskiK** [comment]
Thanks for raising the issue, @pjtaylor. 
It was similar to previously resolved #33, but you've got case where objects sit deeper. 
Fixed and released in ver.0.50.

**Comment by pjtaylor** [comment]
Many thanks for your swift response, @NowinskiK
